portability-gateway/src/main/java/org/dataportabilityproject/gateway/reference/ReferenceApiUtils.java

```diff
@@ -0,0 +1,172 @@
+/*
+* Copyright 2018 The Data-Portability Project Authors.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* https://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+package org.dataportabilityproject.gateway.reference;
+
+import com.google.common.base.Charsets;
+import com.google.common.base.Preconditions;
+import com.google.common.base.Strings;
+import com.google.common.io.BaseEncoding;
+import com.google.common.net.HttpHeaders;
+import com.sun.net.httpserver.Headers;
+import com.sun.net.httpserver.HttpExchange;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.net.HttpCookie;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+import java.util.regex.Pattern;
+import javax.crypto.SecretKey;
+import org.apache.http.NameValuePair;
+import org.apache.http.client.utils.URIBuilder;
+import org.dataportabilityproject.spi.gateway.auth.AuthServiceProviderRegistry.AuthMode;
+import org.simpleframework.http.Cookie;
+import org.simpleframework.http.parse.CookieParser;
+
+/**
+ * Contains utility functions for use by the PortabilityServer HttpHandlers
```

Comment:
s/PortabilityServer/ReferenceApiServer ?

Reply:
ReferenceApiServer, Done.

Reply:
nit: might  as well {@link} it